hcloud/schema_gen.go

```diff
@@ -0,0 +1,875 @@
+// goverter:extend ipNetFromString
+// goverter:extend stringFromIPNet
+// goverter:extend timeToTimePtr
+// goverter:extend serverFromInt64
+// goverter:extend int64FromServer
+// goverter:extend networkFromInt64
+// goverter:extend int64FromNetwork
+// goverter:extend volumeFromInt64
+// goverter:extend int64FromVolume
+// goverter:extend certificateFromInt64
+// goverter:extend int64FromCertificate
+// goverter:extend locationFromString
+// goverter:extend stringFromLocation
+// goverter:extend serverTypeFromInt64
+// goverter:extend int64FromServerType
+// goverter:extend floatingIPFromInt64
+// goverter:extend int64FromFloatingIP
+// goverter:extend mapFromFloatingIPDNSPtrSchema
+// goverter:extend floatingIPDNSPtrSchemaFromMap
+// goverter:extend mapFromPrimaryIPDNSPtrSchema
+// goverter:extend primaryIPDNSPtrSchemaFromMap
+// goverter:extend mapFromServerPublicNetIPv6DNSPtrSchema
+// goverter:extend serverPublicNetIPv6DNSPtrSchemaFromMap
+// goverter:extend firewallStatusFromSchemaServerFirewall
+// goverter:extend serverFirewallSchemaFromFirewallStatus
+// goverter:extend durationFromIntSeconds
+// goverter:extend intSecondsFromDuration
+// goverter:extend serverFromImageCreatedFromSchema
+// goverter:extend anyFromLoadBalancerType
+// goverter:extend serverMetricsTimeSeriesFromSchema
+// goverter:extend loadBalancerMetricsTimeSeriesFromSchema
+// goverter:extend stringPtrFromLoadBalancerServiceProtocol
+// goverter:extend stringPtrFromNetworkZone
+// goverter:extend schemaFromLoadBalancerCreateOptsTargetLabelSelector
+// goverter:extend schemaFromLoadBalancerCreateOptsTargetServer
+// goverter:extend schemaFromLoadBalancerCreateOptsTargetIP
+// goverter:extend stringMapToStringMapPtr
+type converter interface {
+
+	// goverter:map Error.Code ErrorCode
+	// goverter:map Error.Message ErrorMessage
+	ActionFromSchema(schema.Action) *Action
+
+	// goverter:map . Error | schemaActionErrorFromAction
+	SchemaFromAction(*Action) schema.Action
+
+	ActionsFromSchema([]schema.Action) []*Action
+
+	SchemaFromActions([]*Action) []schema.Action
+
+	// goverter:map . IP | ipFromFloatingIPSchema
+	// goverter:map . Network | networkFromFloatingIPSchema
+	FloatingIPFromSchema(schema.FloatingIP) *FloatingIP
+
+	// goverter:map . IP | floatingIPToIPString
+	SchemaFromFloatingIP(*FloatingIP) schema.FloatingIP
+
+	// goverter:map . IP | ipFromPrimaryIPSchema
+	// goverter:map . Network | networkFromPrimaryIPSchema
+	PrimaryIPFromSchema(schema.PrimaryIP) *PrimaryIP
+
+	// goverter:map . IP | primaryIPToIPString
+	SchemaFromPrimaryIP(*PrimaryIP) schema.PrimaryIP
+
+	ISOFromSchema(schema.ISO) *ISO
+
+	SchemaFromISO(*ISO) schema.ISO
+
+	LocationFromSchema(schema.Location) *Location
+
+	SchemaFromLocation(*Location) schema.Location
+
+	DatacenterFromSchema(schema.Datacenter) *Datacenter
+
+	SchemaFromDatacenter(*Datacenter) schema.Datacenter
+
+	ServerFromSchema(schema.Server) *Server
+
+	// goverter:map OutgoingTraffic OutgoingTraffic | mapZeroUint64ToNil
```

Comment:
`goverter:map` supports omitting the source name if they are equal to the target name.
```
	// goverter:map OutgoingTraffic | mapZeroUint64ToNil
```